extensions/jcache/deployment/src/main/java/org/apache/camel/quarkus/component/jcache/deployment/JcacheProcessor.java

```diff
@@ -34,13 +38,20 @@ FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
-    /**
-     * Remove this once this extension starts supporting the native mode.
-     */
-    @BuildStep(onlyIf = NativeBuild.class)
-    @Record(value = ExecutionTime.RUNTIME_INIT)
-    void warnJvmInNative(JvmOnlyRecorder recorder) {
-        JvmOnlyRecorder.warnJvmInNative(LOG, FEATURE); // warn at build time
-        recorder.warnJvmInNative(FEATURE); // warn at runtime
+    @BuildStep
+    void nativeResources(BuildProducer<ServiceProviderBuildItem> services,
+            BuildProducer<NativeImageResourceBuildItem> nativeResources) {
```

Comment:
It seems that `nativeResources` does not be used in the method. Did you want to regiter some resouces in native image?

Reply:
@ppalaga suggested this, see here: https://github.com/apache/camel-quarkus/issues/1846#issuecomment-1245107767 - the purpose is to discover caching providers.

Reply:
OK, I get it. So it looks like only need `BuildProducer<ServiceProviderBuildItem> services` here. Is it right? @ppalaga

Reply:
Indeed, the `nativeResources` method param looks not to be used. Also, I wonder whether methods like `ServiceProviderBuildItem.allProvider*` could help here.

Reply:
Yes, `BuildProducer<NativeImageResourceBuildItem> nativeResources` are not necessary. Sorry for the confusion.

Reply:
Done, removed